### ISSUE TITLE
LuaRocks should use pkg-config

### ISSUE THREAD

**Issue by albertz** [keplerproject/luarocks#514]
In some cases, the correct way to find a package (at least on Linux/Unix) is only via `pkg-config`.
I have such a case in https://github.com/zeromq/lzmq/issues/45.

I found other Lua packages (e.g. [here](https://github.com/deepakjois/luaharfbuzz)) which do this somehow in a indirect way by using a Makefile but I think LuaRocks should automatically use `pkg-config` (at least on Linux/Unix).

**Comment by teto** [comment]
yeah pkg-config usage would be awesome: we have issues dealing with dependencies on nix because of the shortcomings see:
- https://github.com/luarocks/luarocks/issues/766
- https://github.com/NixOS/nixpkgs/pull/61818
So we resort to specifying the dependencies manually sometime which is a pain
